In [1]:
!pip install mlxtend

In [2]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
from mlxtend.feature_selection import SequentialFeatureSelector as SFS



In [3]:
path = r"C:\Users\Admin\Desktop\DATA SCIENCE\DatasetsGame\Video_Games.csv"

DATA = pd.read_csv(path, low_memory=False)
DATA = DATA.dropna(subset=['Year_of_Release'])
DATA['Critic_Score'].fillna(0, inplace=True)
DATA['Year_of_Release'] = pd.to_datetime(DATA['Year_of_Release'], format='%Y', errors='coerce').dt.year
df = DATA[['Name', 'Platform', 'Year_of_Release', 'Genre', 'Publisher', 'NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales', 'Critic_Score']]

df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii             2006        Sports   
1          Super Mario Bros.      NES             1985      Platform   
2             Mario Kart Wii      Wii             2008        Racing   
3          Wii Sports Resort      Wii             2009        Sports   
4   Pokemon Red/Pokemon Blue       GB             1996  Role-Playing   
5                     Tetris       GB             1989        Puzzle   
6      New Super Mario Bros.       DS             2006      Platform   
7                   Wii Play      Wii             2006          Misc   
8  New Super Mario Bros. Wii      Wii             2009      Platform   
9                  Duck Hunt      NES             1984       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  
0          76.0  
1           0.0  
2          82.0  
3          80.0  
4           0.0  
5           0.0  
6          89.0  
7          58.0  
8          87.0  
9           0.0

In [11]:
X = df[['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales']]
Y = df['Global_Sales']

In [12]:

X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2)
model_ = LinearRegression()
sfs = SFS(model_,
         k_features=(1, len(X.columns)),
           forward=False,
           floating=False,
           scoring='neg_mean_squared_error',
           cv=5)

sfs.fit(X_train, y_train)
features = list(sfs.k_feature_names_)
print("Features seleccionados: ", features)

model = LinearRegression()
model.fit(X_train[features], y_train)
y_pred = model.predict(X_test[features])
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print("Error cuadratico medio(RMSE): ", mse)
print("Aptitud del modelo(R^2): ", r2)

Features seleccionados:  ['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales']
Error cuadratico medio(RMSE):  2.7731432800565317e-05
Aptitud del modelo(R^2):  0.9999863557674393


En conclusion, el modelo nos dio unos buenos resultados de prueba. Utilice columnas que ya sabia estaban correlacionadas, siendo las de ventas segun region y la global.
En cuanto a los resultados, en el error cuadratico medio obtuvimos un resultado bajo, indicando que los puntos de los datos observados estan cerca de los puntos predichos. Mientras que para el R^2, obtuvimos que la aptitud del modelo es casi perfecta con un 0.9999863557674393 de 1.
